src/dune_rules/pkg_rules.ml

```diff
@@ -1247,26 +1263,40 @@ end = struct
 
   let resolve =
     let module Input = struct
-      type t = DB.t * Context_name.t * Package.Name.t
+      type t = DB.t * Context_name.t * Package.Name.t * string list
```

Comment:
This would be clearer if a more meaningful type than `string list` was used to represent the components. Maybe it's time to use a record type rather than a giant tuple.